File: QuanLyPhongKham.Server/src/QuanLyPhongKham.DTOs/Requests/ChiTietDonThuocRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.DTOs.Requests
{
    public class ChiTietDonThuocRequest
    {
        public Guid? MaCTDT { get; set; }

        [Required(ErrorMessage = "Mã thuốc là bắt buộc.")]
        public Guid MaThuoc { get; set; }

        [Required(ErrorMessage = "Số lượng là bắt buộc.")]
        [Range(1, 9999, ErrorMessage = "Số lượng phải từ 1 đến 9999.")]
        public int SoLuong { get; set; }

        [MaxLength(500, ErrorMessage = "Liều dùng không vượt quá 500 ký tự.")]
        public string? LieuDung { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/interfaces/requests/ChiTietDonThuocRequest.js
/**
 * @typedef {Object} ChiTietDonThuocRequest
 * @property {string} maCTDT
 * @property {string} maThuoc
 * @property {number} soLuong
 * @property {string} lieuDung
 */

export const createChiTietDonThuocRequest = (
  maCTDT = "",
  maThuoc = "",
  soLuong = 1,
  lieuDung = "",
) => {
  const request = {
    maThuoc,
    soLuong,
    lieuDung,
  };
  if (maCTDT) request.maCTDT = maCTDT;
  return request;
};
